app/lib/actions/katello/repository/destroy.rb

```diff
@@ -14,7 +14,7 @@ def plan(repository, options = {})
           action_subject(repository)
           plan_action(ContentViewPuppetModule::Destroy, repository) if repository.puppet?
           plan_action(Pulp::Repository::Destroy, pulp_id: repository.pulp_id)
-          plan_action(Product::ContentDestroy, repository) unless skip_environment_update
+          plan_action(Product::ContentDestroy, repository)
```

Comment:
content has a many-to-many relationship with an org and can exist when associated to no org.  So we cannot rely on org delete in candlepin to clean up content, in fact that would be bad if it did, as the redhat content is shared between orgs